File: 프로그래머스/2/42578. 의상/의상.js
function solution(clothes) {
    const closet = {}
    
    for (const [item, type] of clothes){
        if(!closet[type]) closet[type] = 0;
        closet[type] ++;
    }
    
    let answer = 1;
    for (const type in closet){
        answer *= (closet[type] + 1);
    }
    
    return answer -1;
}

File: 프로그래머스/2/42578. 의상/의상.java
import java.util.*;
class Solution {
    public int solution(String[][] clothes) {
        Map<String, Integer> closet = new HashMap<>();
        
        for (String[] cloth: clothes){
            String type = cloth[1];
            closet.put(type, closet.getOrDefault(type,0) + 1);
        }
        
        int answer = 1;
        for (int count: closet.values()){
            answer *= (count + 1);
        }
        
        return answer - 1;
    }
}